### PR TITLE
Pytorch backend PODMIONet add three branch connection method:"mul","sum","cat"; and add optional port net after connection

### PR DESCRIPTION
**Pull request by Jerry-Jzy**
## what is the purpose of this request?
Pytorch backend PODMIONet add three branch connection method:"mul","sum","cat"; and add optional port net after connection.

### CODE REVIEW

File: deepxde/nn/pytorch/mionet.py [hunk 1]
```diff
@@ -89,18 +89,22 @@ def __init__(
         pod_basis,
         layer_sizes_branch1,
         layer_sizes_branch2,
+        layer_sizes_port,
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:92]: `layer_sizes_port=None,` --- by default no net

Comment (Jerry-Jzy) [reply deepxde/nn/pytorch/mionet.py]: Fixed

File: deepxde/nn/pytorch/mionet.py [hunk 2]
```diff
@@ -91,16 +91,20 @@ def __init__(
         layer_sizes_branch2,
         activation,
         kernel_initializer,
+        connect_method="mul",
         layer_sizes_trunk=None,
         regularization=None,
         trunk_last_activation=False,
+        layer_sizes_merge=None,
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:98]: merge --> merger

File: deepxde/nn/pytorch/mionet.py [hunk 3]
```diff
@@ -91,16 +91,20 @@ def __init__(
...
+        merge_net_after_branch=False,
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:99]: You don't need this. If layer_sizes is None, then there is False.

File: deepxde/nn/pytorch/mionet.py [hunk 4]
```diff
@@ -91,16 +91,20 @@ def __init__(
         layer_sizes_branch2,
         activation,
         kernel_initializer,
+        connect_method="mul",
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:94]: `merge_operation="mul",`

File: deepxde/nn/pytorch/mionet.py (hunk 2 above)
Comment (Jerry-Jzy) [reply deepxde/nn/pytorch/mionet.py]: Fixed

File: deepxde/nn/pytorch/mionet.py (hunk 3 above)
Comment (Jerry-Jzy) [reply deepxde/nn/pytorch/mionet.py]: Fixed

File: deepxde/nn/pytorch/mionet.py (hunk 4 above)
Comment (Jerry-Jzy) [reply deepxde/nn/pytorch/mionet.py]: Fixed

File: deepxde/nn/pytorch/mionet.py [hunk 5]
```diff
@@ -91,16 +91,19 @@ def __init__(
         layer_sizes_branch2,
         activation,
         kernel_initializer,
+        merge_operation="mul",
         layer_sizes_trunk=None,
         regularization=None,
         trunk_last_activation=False,
+        layer_sizes_merger=None
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:98]: Format via black.

File: deepxde/nn/pytorch/mionet.py [hunk 6]
```diff
@@ -138,12 +153,28 @@ def forward(self, inputs):
...
+        if self.merge_operation == "cat":
+            x_merger = torch.cat((y_func1, y_func2), 1)
+        else:
+            if y_func1.shape[-1] != y_func2.shape[-1]:
+                raise AssertionError(
+                    "Output sizes of branch1 net and branch2 net do not match."
+                )
+            if self.merge_operation == "sum":
+                x_merger = y_func1 + y_func2
+            elif self.merge_operation == "mul":
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:166]:
```
if self.merge_operation == "mul":
    ...
elif self.merge_operation == "cat":
    ....
elif self.merge_operation == "sum":
    ....
```

Comment (Jerry-Jzy) [reply deepxde/nn/pytorch/mionet.py:166]: Do I need to care about the size issue for "mul" and "sum"?

File: deepxde/nn/pytorch/mionet.py [hunk 7]
```diff
@@ -91,16 +91,19 @@ def __init__(
         layer_sizes_branch2,
         activation,
         kernel_initializer,
+        merge_operation="mul",
```
Comment (lululxvi) [inline deepxde/nn/pytorch/mionet.py:94]: Move this to the second last positoin

Comment (Jerry-Jzy) [reply deepxde/nn/pytorch/mionet.py]: Fixed

**Pull request merged**